### PR TITLE
Adds a reusable lambda component

### PR DESCRIPTION
**Pull request by JakeGinnivan**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{
```
Comment (thestillheron) [inline .eslintrc.json:1]: There's a slight irony that you updated the tab spaces to be 4, but this new file is still at tabspace 2 :P

File: libs/pulumi-lambda/src/lib/deny-log-group-create.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    constructor(
+        name: string,
+        // eslint-disable-next-line @typescript-eslint/ban-types
+        _: {},
+        opts?: pulumi.ComponentResourceOptions | undefined,
+    ) {
+        super('wanews:deny-log-group-create-policy', name, {}, opts)
+
+        this.policy = new aws.iam.Policy('deny-log-group-creation', {
```
Comment (thestillheron) [inline libs/pulumi-lambda/src/lib/deny-log-group-create.ts:15]: I'm still not completely across the rules on when to use it and when not to, but is this missing:
`{ parent: this }`

**Review by thestillheron — APPROVED** [review]
Possibly a missing `{ parent: this }` but everything else seems good!

File: .eslintrc.json (hunk 1 above)
Comment (thestillheron) [reply .eslintrc.json:1]: In fact (possibly as a separate PR) it seems like a whole-project prettier run might be in order.

Comment (bennettp123) [reply .eslintrc.json:1]: tabspace 2 forever! 😈

**Review by bennettp123 — APPROVED** [review]
just the missing `{ parent: this }`, as previously mentioned by @thestillheron

File: libs/pulumi-lambda/src/lib/deny-log-group-create.ts (hunk 2 above)
Comment (JakeGinnivan) [reply libs/pulumi-lambda/src/lib/deny-log-group-create.ts:15]: Good catch, should always have `parent: this` inside a ComponentResource

**Pull request merged**